src/Explorer/Notebook/NotebookComponent/ContentProviders/InMemoryContentProvider.ts

```diff
@@ -0,0 +1,119 @@
+import { FileType, IContent, IContentProvider, IGetParams, ServerConfig } from "@nteract/core";
+import { Observable, of } from "rxjs";
+import { AjaxResponse } from "rxjs/ajax";
+import { HttpStatusCodes } from "../../../../Common/Constants";
+import { getErrorMessage } from "../../../../Common/ErrorHandlingUtils";
+import * as Logger from "../../../../Common/Logger";
+
+export interface InMemoryContentProviderParams {
+  [path: string]: { readonly: boolean; content: IContent<FileType> };
+}
+
+class InMemoryContentProviderError extends Error {
+  constructor(error: string, public errno: number = InMemoryContentProvider.SelfErrorCode) {
+    super(error);
+  }
+}
```

Comment:
This class is faking Ajax responses even though it is not doing any networking. In case of error Nteract expects `errno` property to be present in the error object to infer what kind of error occurred and make decisions. That is the only reason I have a wrapper around `Error`. We do something similar in `GitHubContentProvider`.

Unfortunately Nteract's `IContentProvider` interface only understands Ajax responses and we would at some point need to update it to be more generic so that we don't need to create fake Ajax responses in scenarios like ours.

Reply:
I'm adding a comment explaining this too.